### PR TITLE
fix: traces UI fixes

### PR DESCRIPTION
**Pull request by omkarK06**
1. Removed basic and advanced filter modes. Added advanced as default mode
2. In trace details, removed inline span expand option option

## Release Notes

- **New Features**
  - Added a "Links" tab in the Trace Details Sidebar for better navigation between span links.
  - Introduced a new utility function for converting time from nanoseconds to milliseconds.

- **Improvements**
  - Simplified the trace filtering process by removing unnecessary filter type selections and logic.
  - Enhanced error message rendering for improved clarity, particularly when no stream or trace data is found.
  - Improved the layout and styling of the Trace Details Sidebar for better user experience.
  - Enhanced the display of span details, including dynamic visibility of span operation names and additional computed properties.
  - Streamlined the rendering logic in various components for better readability and user interaction.
  - Updated the handling of URL query parameters for improved state management.

- **Bug Fixes**
  - Enhanced error handling when no trace details are returned.

These updates streamline the user interface and improve functionality, making the application more intuitive and user-friendly.

### CODE REVIEW

File: web/src/plugins/traces/IndexList.vue [hunk 1]
```diff
@@ -98,63 +98,7 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
-                      (currValue, prevValue) =>
-                        updateQueryFilter(props.row.name, currValue, prevValue)
-                    "
-                    @update:search-keyword="getFieldValues(props.row.name)"
-                  />
-                </template>
-                <template v-else>
-                  <basic-values-filter :row="props.row" />
-                </template>
+                <basic-values-filter :row="props.row" />
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/traces/IndexList.vue]: Done

File: web/src/utils/zincutils.ts [hunk 2]
```diff
@@ -922,3 +922,10 @@ export const getTimezonesByOffset = async (offsetMinutes: number) => {
     return zoneOffset === offsetHours;
   });
 };
+
+export const converTimeFromNsToMs = (time: number) => {
+  const nanoseconds = time;
+  const milliseconds = Math.floor(nanoseconds / 1000000);
+  const date = new Date(milliseconds);
+  return date.getTime();
+};
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/utils/zincutils.ts:931]: If we divide directly it returns timestamp in decimal

File: web/src/plugins/traces/TraceDetailsSidebar.vue [hunk 3]
```diff
@@ -544,6 +651,56 @@ export default defineComponent({
...
+      router.push({
+        name: "traceDetails",
+        query,
+      });
+
+      emit("open-trace");
+    };
+
+    const spanLinks = computed(() => JSON.parse(props.span.links));
+
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/traces/TraceDetailsSidebar.vue]: Done

File: web/src/plugins/traces/TraceDetailsSidebar.vue [hunk 4]
```diff
@@ -544,6 +651,56 @@
       navigateToLogs(queryDetails);
     };
 
+    const getStartTime = computed(() => {
+      return (
+        converTimeFromNsToMs(props.span.start_time) -
+        props.baseTracePosition.startTimeMs +
+        "ms"
+      );
+    });
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/traces/TraceDetailsSidebar.vue:660]: const getStartTime = computed(() => {
      return (
        converTimeFromNsToMs(props.span.start_time) -
        (props.baseTracePosition?.startTimeMs || 0) +
        "ms"
      );
    });

File: web/src/plugins/traces/TraceDetailsSidebar.vue [hunk 5]
```diff
@@ -30,42 +35,62 @@
...
+        :title="getDuration"
+        style="border-right: 1px solid #cccccc; font-size: 14px"
+      >
+        <span class="text-grey-7">Duration: </span>
+        <span>{{ getDuration }}</span>
+      </div>
+
+      <div
+        class="q-px-sm ellipsis non-selectable"
+        :title="getDuration"
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/traces/TraceDetailsSidebar.vue]: Done

**Review by hengfeiyang — APPROVED** [review]

File: web/src/plugins/traces/TraceDetails.vue [hunk 6]
```diff
@@ -311,7 +311,11 @@ import TraceDetailsSidebar from "./TraceDetailsSidebar.vue";
 import TraceTree from "./TraceTree.vue";
 import TraceHeader from "./TraceHeader.vue";
 import { useStore } from "vuex";
-import { formatTimeWithSuffix, getImageURL } from "@/utils/zincutils";
+import {
+  formatTimeWithSuffix,
+  getImageURL,
+  converTimeFromNsToMs,
+} from "@/utils/zincutils";
```
[the remark this answers is not in the document]
Comment (gaby) [reply web/src/plugins/traces/TraceDetails.vue:318]: @omkarK06 This is probably an issue?

File: web/src/plugins/traces/SearchBar.vue [hunk 7]
```diff
@@ -632,18 +558,17 @@ export default defineComponent({
...
-    font-size: 12px;
-  }
-  .q-icon {
-    margin-right: 4px;
+    :deep(.block) {
+      font-size: 12px;
+    }
+    :deep(.q-icon) {
+      margin-right: 4px;
+    }
```
[the remark this answers is not in the document]
Comment (gaby) [reply web/src/plugins/traces/SearchBar.vue:571]: Same here

File: web/src/plugins/traces/TraceDetails.vue [hunk 8]
```diff
@@ -1086,6 +1094,11 @@ export default defineComponent({
       });
     };
 
+    const openTraceLink = () => {
+      resetTraceDetails();
+      setupTraceDetails();
+    };
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/traces/TraceDetails.vue]: Fixed

File: web/src/plugins/traces/TraceDetails.vue (hunk 6 above)
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/traces/TraceDetails.vue]: @coderabbitai fixed

File: web/src/plugins/traces/SearchBar.vue (hunk 7 above)
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/traces/SearchBar.vue]: Fixed

File: web/src/plugins/traces/TraceDetailsSidebar.vue [hunk 9]
```diff
@@ -436,6 +524,25 @@ export default defineComponent({
       },
     ]);
 
+    const linkColumns = ref([
+      {
+        name: "traceId",
+        field: (row: any) => row.context.traceId,
+        prop: (row: any) => row.context.traceId,
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/traces/TraceDetailsSidebar.vue]: Fixed

File: web/src/plugins/traces/TraceDetailsSidebar.vue [hunk 10]
```diff
@@ -544,6 +651,68 @@
...
+      );
+    });
+
+    const copySpanId = () => {
+      q.notify({
+        type: "positive",
+        message: "Span ID copied to clipboard",
+        timeout: 2000,
+      });
+      copyToClipboard(props.span.span_id);
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/traces/TraceDetailsSidebar.vue]: Fixed

File: web/src/plugins/traces/TraceDetailsSidebar.vue [hunk 11]
```diff
@@ -544,6 +651,68 @@ export default defineComponent({
...
+    };
+
+    const spanLinks = computed(() => {
+      try {
+        return JSON.parse(props.span.links);
+      } catch (e) {
+        console.log("Error parsing span links:", e);
+        return [];
+      }
+    });
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/traces/TraceDetailsSidebar.vue]: Fixed

File: web/src/plugins/traces/SearchBar.vue (hunk 7 above)
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/traces/SearchBar.vue]: @coderabbitai

File: web/src/plugins/traces/SearchBar.vue [hunk 12]
```diff
@@ -275,7 +218,7 @@
       (fields) => {
         if (fields.length) updateFieldKeywords(fields);
       },
-      { immediate: true, deep: true }
+      { immediate: true, deep: true },
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/traces/SearchBar.vue:221]: Required

File: web/src/plugins/traces/SearchBar.vue [hunk 13]
```diff
@@ -186,7 +131,7 @@ export default defineComponent({
   components: {
     DateTime,
     QueryEditor: defineAsyncComponent(
-      () => import("@/components/QueryEditor.vue")
+      () => import("@/components/QueryEditor.vue"),
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/plugins/traces/SearchBar.vue:134]: This is expected

File: web/src/utils/query/sqlUtils.ts [hunk 14]
```diff
@@ -406,18 +406,18 @@ function parseCondition(condition: any) {
         };
       }
     } else if (condition.type === "function") {
-      let conditionName = condition?.name?.name[0]?.value?.toLowerCase();
+      const conditionName = condition?.name?.name[0]?.value?.toLowerCase();
```
[the remark this answers is not in the document]
Comment (omkarK06) [reply web/src/utils/query/sqlUtils.ts:409]: Fixed

**Pull request merged**